src/index.js

```diff
@@ -77,11 +82,7 @@ const mapToPropertyDescriptor = (
         return this.hasAttribute(name);
       },
       set(value) {
-        if (value) {
-          this.setAttribute(name, '');
-        } else {
-          this.removeAttribute(name);
-        }
+        setBooleanAttribute(this, name, value);
```

Comment:
yeah, okay. I kinda thought it was nice to tell immediately if something is pure. But yeah this may be overdoing it.